core/meta/inc/TClass.h

```diff
@@ -429,7 +431,7 @@ friend class TStreamerInfo;
    ROOT::DesFunc_t    GetDestructor() const;
    ROOT::DelArrFunc_t GetDeleteArray() const;
    ClassInfo_t       *GetClassInfo() const {
-      if (fCanLoadClassInfo && !TestBit(kLoading))
```

Comment:
Could you explain why we do not need that condition any more?

Reply:
it is moved to `TClass.cxx:5830` and 'explained' in the commit log: [`0f9523f` (#15113)](https://github.com/root-project/root/pull/15113/commits/0f9523faeb10cd28a351b99577fb948570a14e49)

Reply:
AFAICT you are linking to the current commit 0f9523faeb10cd28a351b99577fb948570a14e49, which only has the summary "TClass::GetClassInfo Move TestBit behind lock". Where would I look for the explanation?

Reply:
> Could you explain why we do not need that condition any more?

We still need it.  It is moved inside the function call.   It is now behind/after the lock taking to avoid any concurrent access to `fBits`.   The details of the related complaint are at https://github.com/root-project/root/issues/15090

Reply:
I have added an expanded explanation in the commit.